allen-cell-animated/timelapse-colorizer
# feat: Added progress circle for the dataset load

Problem
=======
Closes #403, show progress bar instead of loading spinner.

*Estimated review size: small, 15 minutes*

Solution
========
- Updates `LoadingSpinner` to accept an integer percent `progress`. When set, it replaces the loading spinner with the progress bar.
- Updates `Dataset.open` to report loading status via callbacks.
  - These are captured by `Collection` and reported up to the `Viewer`, which stores it as state and passes it into `CanvasWrapper` which controls the loading spinner.
- Moves old console statements out of `Dataset` so they don't trigger during unit testing.

## Type of change

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-404/
2. Click on any of the provided datasets. (The exploratory dataset is the largest.)
3. The progress bar should appear as the dataset loads.
4. Switch datasets from the dropdown.

Screenshots (optional):
-----------------------
![image](https://github.com/allen-cell-animated/timelapse-colorizer/assets/30200665/911d4067-e586-47a3-9f57-643ded6e63a0)

Keyfiles (delete if not relevant):
-----------------------
1. `Dataset.tsx`
5. `Viewer.tsx`
6. `LoadingSpinner.tsx`

Thanks for contributing!
